js/ui/animation.js

```diff
@@ -30,11 +31,10 @@ var Animation = new Lang.Class({
 
     play: function() {
         if (this._isLoaded && this._timeoutId == 0) {
-            if (this._frame == 0)
-                this._showFrame(0);
-
-            this._timeoutId = GLib.timeout_add(GLib.PRIORITY_LOW, this._speed, Lang.bind(this, this._update));
-            GLib.Source.set_name_by_id(this._timeoutId, '[gnome-shell] this._update');
+            // Set the frame to be the last one, so when we update it
+            // the first time, the first frame will be displayed.
+            this._frame = this._animations.get_n_children() - 1;
```

Comment:
Is this always what you want? It's a behavior change from how it worked previously, where if you called `stop()` and then again `play()` it would start again from the frame it stopped at. Now you're always resetting it instead.